lib/WebSocket.js

```diff
@@ -262,10 +262,10 @@ class WebSocket extends EventEmitter {
    */
   close (code, data) {
     if (this.readyState === WebSocket.CLOSED) return;
-
     if (this.readyState === WebSocket.CONNECTING) {
-      this.readyState = WebSocket.CLOSED;
-      return;
+      this._req.abort();
+      this.emit('error', new Error('closed while connecting'));
```

Comment:
Ok, I'll change it to that, merge this and release `2.0.0-beta.0` if there are no objections.